File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = 0
    stack = []
    
    for char in s:
        if stack and stack[-1] == char: # stack에 값이 있고, 마지막 값이 현재 값이랑 같으면
            stack.pop()
        else :
            stack.append(char)
    
    if stack:
        answer = 0
    else :
        answer = 1
    return answer

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution
{
    public int solution(String s)
    {
        int answer = -1;
        char[] arr = s.toCharArray();
        Stack<Character> stack = new Stack<>();
        
        for (char ch : arr) {
            if (!stack.isEmpty() && stack.peek() == ch) {
                stack.pop();
                continue;
            }
            
            stack.push(ch);
        }
        
        answer = stack.isEmpty() ? 1 : 0;
        return answer;
    }
}